k8s-install/scripts/install-cni.sh

```diff
@@ -201,17 +201,17 @@ echo "Created CNI config ${CNI_CONF_NAME}"
 # This prevents Kubernetes from restarting the pod repeatedly.
 should_sleep=${SLEEP:-"true"}
 echo "Done configuring CNI.  Sleep=$should_sleep"
-while [ "$should_sleep" == "true"  ]; do
+while [ "$should_sleep" = "true"  ]; do
   # Kubernetes Secrets can be updated.  If so, we need to install the updated
   # version to the host. Just check the timestamp on the certificate to see if it
   # has been updated.  A bit hokey, but likely good enough.
-  if [ -e ${SECRETS_MOUNT_DIR}/etcd-cert ];
+  if [ -e "${SECRETS_MOUNT_DIR}"/etcd-cert ];
   then
-    stat_output=$(stat -c%y ${SECRETS_MOUNT_DIR}/etcd-cert 2>/dev/null)
+    stat_output=$(stat -c%y "${SECRETS_MOUNT_DIR}"/etcd-cert 2>/dev/null)
     sleep 10;
-    if [ "$stat_output" != "$(stat -c%y ${SECRETS_MOUNT_DIR}/etcd-cert 2>/dev/null)" ]; then
+    if [ "$stat_output" != "$(stat -c%y "${SECRETS_MOUNT_DIR}"/etcd-cert 2>/dev/null)" ]; then
       echo "Updating installed secrets at: $(date)"
-      cp -p ${SECRETS_MOUNT_DIR}/* /host/etc/cni/net.d/calico-tls/
+      cp -p "${SECRETS_MOUNT_DIR}"/* /host/etc/cni/net.d/calico-tls/
```

Comment:
So this started out because once I switched to Debian from Alpine a bunch of the lines in this shell script started breaking.

Things like conditional equality checks using `==` failed. 

So on a suggestion from Jonathan P. I downloaded [`shellcheck`](https://github.com/koalaman/shellcheck) which does static analysis on shell scripts to spot any other bad practices. 

That lead to 90% of the small fixes above that you see.

Will be great when we eventually move to using a Go binary 😅.